### DISCUSSION TITLE
Recordings as fragmented MP4 files take a long time to load in the Chrome browser

### DISCUSSION THREAD

**Discussion by fabiox23** [bluenviron/mediamtx#3906] [Questions]
### Question

As the title says, when a recorded stream (formatted as an fragmented MP4 video file and located on a remote machine) is viewed through a web browser, it takes a very long time (10-20 seconds) for the video to load in Chrome whilst less than a second in Firefox. From the console, I can see Chrome is downloading a large number of video fragments.

The behaviour is the same both when the mp4 file is viewed directly (ie URL ending with mp4) and when viewed through a simple html file with a video element (including when using a library such as videojs). The video has audio.

I have tried to search for the issue on the web and with ChatGPT to no avail.

Is it possible to tweak the html file so that Chrome behaves nicely and loads the video quickly, just as Firefox?

Thanks.

**Comment by fabiox23** [comment]
Can anyone help with this issue, please?

**Comment by aler9** [comment]
Hello, segments are not meant to be as compatible as possible but to be as robust as possible in case of crashes. The playback server is the component in charge of converting segments on disk into something as compatible as possible.

Therefore, enable the playback server, use the playback server to view recordings and also append `format=mp4` at the end of the request to broaden compatibility, as explained in the README:

> the fMP4 format may offer limited compatibility with some players. To fix the issue, it's possible to use the standard MP4 format, by adding format=mp4 to a /get request:

http://localhost:9996/get?path=[mypath]&start=[start_date]&duration=[duration]&format=mp4
